lib/Sema/TypeCheckAttr.cpp

```diff
@@ -3816,8 +3848,20 @@ void AttributeChecker::visitDerivativeAttr(DerivativeAttr *attr) {
   auto vectorTy = valueResultConf.getTypeWitnessByName(
       valueResultType, Ctx.Id_TangentVector);
 
+  // Compute the actual differential/pullback type that we use for comparison
```

Comment:
Note: this code is copied from https://github.com/apple/swift/pull/28762 - it's needed to fix TF-1062. https://github.com/apple/swift/pull/28762 should be merged first!